lib/SIL/SILVerifier.cpp

```diff
@@ -751,7 +751,8 @@ class SILVerifier : public SILVerifierBase<SILVerifier> {
 
     // If there are substitutions, verify them and apply them to the callee.
     if (subs.empty()) {
-      require(!fnTy->isPolymorphic(),
+      require(!fnTy->isPolymorphic() ||
+                  fnTy->getGenericSignature()->areAllParamsConcrete(),
```

Comment:
When I added concrete extensions my intent was that we would never build a SILFunctionType with a fully concrete generic signature. If this invariant is broken do you mind fixing it at the source?